### PR TITLE
Update framework folder structure - Closes #5689

### LINKED ISSUE #5689: Framework folder structure should follow class structure
### Description
Framework structure should follow the logical structure
It should be as
```
/src
 - application.ts
 - errors.ts
 - system_dir.ts
 ...
 - logger/
 - controller/
 - plugins/
 - modules/
 - schema/
 - node/
   - forger/
   - network/
   ...
   - node.ts
   - index.ts
```
Unit test structure should follow the same.
Also, unit test structure should be the same without `spec` folder

### Acceptance criteria
- It should not contain any code change except `import` path changes and test settings
- All build should pass
- All test should pass

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5689 

### How was it solved?

- Only moving the folder structure and fixing import path

### How was it tested?

All build and test should work

### CODE REVIEW

**Review by nazarhussain — CHANGES_REQUESTED** [review]
Looks good, except few small suggestions. 

- Rename `systems_dirs.ts` to `utils.ts`
- Merge `src/controller/types.ts` to `src/types.ts`
- Move `src/controller/constants.ts`  to `src/constants.ts`

File: framework/src/application.ts [hunk 1]
```diff
@@ -215,7 +209,7 @@ export class Application {
 		this.logger.info(
 			'Contribution guidelines can be found at Lisk-docs: https://github.com/LiskHQ/lisk-docs/blob/build/CONTRIBUTING.adoc',
```
Comment (nazarhussain) [inline framework/src/application.ts:210]: Checked accidentally, seems this link is wrong. Its referring to contribution to the documentation not the sdk. SDK contribution document is https://github.com/LiskHQ/lisk-sdk/blob/development/docs/CONTRIBUTING.md

**Review by nazarhussain — CHANGES_REQUESTED** [review]

**Review by ManuGowda — COMMENTED** [review]
LGTM, functional test dir structure doesn't follow new structure.

**Comment by shuse2** [comment]
@ManuGowda I think it is following the structure, please recheck again.
@nazarhussain  I think `systems_dirs` is explicit what it has. If we include more function we can consider to rename it accordingly.

**Review by nazarhussain — APPROVED** [review]

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**